ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/BackupReplayFileTest.java

```diff
@@ -80,6 +94,62 @@ public void shouldWriteRecord() throws IOException {
     ));
   }
 
+  @Test
+  public void shouldWriteMultipleRecords() throws IOException {
+    // Given
+    final ConsumerRecord<byte[], byte[]> record1= newStreamRecord("stream1");
+    final ConsumerRecord<byte[], byte[]> record2 = newStreamRecord("stream2");
+
+    // When
+    replayFile.write(record1);
+    replayFile.write(record2);
+
+    // Then
+    final List<String> commands = Files.readAllLines(internalReplayFile.toPath());
```

Comment:
Is there a way to confirm that the backup/dirty is also up to date?

Reply:
I suppose we could, but i'd rather not assert on the internal details of the implementation. This is basically what i was trying to capture with a test case that does multiple writes. If the backup were not updated, then we would not see the second write succeed.